api/context.go

```diff
@@ -234,6 +238,47 @@ func (h handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func (cw *CorsWrapper) ServeHTTP(
+	w http.ResponseWriter,
+	r *http.Request) {
+	allowedMethods := []string{
+		"POST",
+		"GET",
+		"OPTIONS",
+		"PUT",
+		"PATCH",
+		"DELETE",
+	}
+
+	allowedHeaders := []string{
+		"Accept",
+		"Content-Type",
+		"Content-Length",
+		"Accept-Encoding",
+		"Authorization",
+		"X-CSRF-Token",
+		"X-Auth-Token",
+	}
+
+	if len(*utils.Cfg.ServiceSettings.AllowCorsFrom) > 0 {
+		w.Header().Set("Access-Control-Allow-Origin", *utils.Cfg.ServiceSettings.AllowCorsFrom)
+
+		w.Header().Set(
+			"Access-Control-Allow-Methods",
```

Comment:
Does `Access-Control-Allow-Methods` and `Access-Control-Allow-Headers` need to be set on every request?  or just on  `OPTIONS` requests.  Sorry not sure how those work in conjunction with `Access-Control-Allow-Origin`